src/SFA.DAS.ApprenticeFeedback.Application.UnitTests/Queries/WhenRequestingFeedbackTargetsForUpdate.cs

```diff
@@ -433,22 +414,72 @@ ApplicationSettings appSettings
             context.SaveChanges();
 
             var query = new GetApprenticeFeedbackTargetsForUpdateQuery() { BatchSize = (int)FeedbackTargetTestData[testCaseIndex].Arguments[1] };
+            var handler = GetHandler(context);
 
-            var handler = new GetApprenticeFeedbackTargetsForUpdateQueryHandler(context, 
-                new ApplicationSettings() 
-                { 
+            // Act
+
+            var result = await handler.Handle(query, CancellationToken.None);
+
+            // Assert
+            var expectedResult = (IEnumerable<Domain.Entities.ApprenticeFeedbackTarget>)FeedbackTargetTestData[testCaseIndex].ExpectedResult;
+            result.ApprenticeFeedbackTargets.Select(aft => aft.ProviderName).Should().BeEquivalentTo(expectedResult.Select(aft => aft.ProviderName));
+        }
+
+        [TestCase(0)]
+        [TestCase(1)]
+        [TestCase(2)]
+        [TestCase(3)]
+        [TestCase(4)]
+        [TestCase(5)]
+        public async Task Then_Only_The_Expected_Number_Of_FeedbackTargetsAreReturned(int batchSize)
+        {
```

Comment:
This was part of the other test. Here I have made it into a separate test, and added the following changes:

- Adjusted the assert to account for the fact that `DbSet.AddRange()` does not guarantee the order of insertion in .NET 6
- Renamed the test to accurately reflect its intentions
- Added test cases for multiple batch sizes. Previously it was using a fixed batch size of 3
- Improve the clarity of the assertion statement